packages/account/src/Sections/Security/LoginHistory/__test__/login-history.spec.tsx

```diff
@@ -0,0 +1,152 @@
+                    status: '',
+                    ip: '',
+                    id: 0,
+                },
+            ],
+        },
+        isLoading: false,
+        isError: false,
+        error: '',
+    };
+
+    const renderComponent = () => {
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <APIProvider>
+                <StoreProvider store={mock_store}>{children}</StoreProvider>
+            </APIProvider>
+        );
+        mockUseLoginHistory.mockReturnValue(response);
+        render(<LoginHistory />, { wrapper });
+    };
+
+    beforeEach(() => {
+        (isMobile as jest.Mock).mockReturnValue(false);
+        mock_store = mockStore({
+            client: {
+                is_switching: false,
+                is_authorize: true,
+            },
+            ui: {
+                is_mobile: false,
+            },
+        });
+
+        response = {
+            data: {
+                formatted_data: [
+                    {
+                        date: '2023-08-28 03:11:45 GMT',
+                        browser: '"Chrome  v116.0.0.0"',
+                        action: 'login',
+                        status: 'successful',
+                        ip: '175.143.37.57',
+                        id: 0,
+                    },
+                ],
+            },
+            isLoading: false,
+            isError: false,
+            error: 'this is an error message',
+        };
+    });
+
+    it('should render Login History List when isMobile is true', () => {
+        mock_store.ui.is_mobile = true;
+        renderComponent();
+        expect(screen.getByText(/date and time/i)).toHaveClass('dc-text login-history__list__row__cell--title');
+    });
+
+    it('should render Login History Table', () => {
+        renderComponent();
+        expect(screen.getByText(/date and time/i)).not.toHaveClass('dc-text login-history__list__row__cell--title');
+    });
+
+    it('should render Table Header.', () => {
+        renderComponent();
+        const table_headers = [/date and time/i, /action/i, /browser/i, /ip address/i, /status/i];
+        table_headers.forEach(header => {
+            expect(screen.getByText(header)).toBeInTheDocument();
+        });
+    });
+
+    it('should render Table Items.', () => {
+        renderComponent();
+        const table_items = [
+            /2023-08-28 03:11:45 GMT/i,
+            /login/i,
+            /chrome v116.0.0.0/i,
+            /175.143.37.57/i,
+            /successful/i,
+        ];
+        table_items.forEach(item => {
+            expect(screen.getByText(item)).toBeInTheDocument();
+        });
+    });
+
+    it('should render Loading if client: is_switching is true', () => {
+        mock_store.client.is_switching = true;
+        renderComponent();
+        expect(screen.getByText('mockedLoading')).toBeInTheDocument();
+    });
+
+    it('should render Loading with classname account__initial-loader if isLoading is true', () => {
```

Comment:
we need to change the description here. Wea re not checking the classname here in this test case